pkg/istio-agent/xds-proxy.go

```diff
@@ -340,50 +375,78 @@ func buildUpstreamClientDialOpts(sa *Agent) ([]grpc.DialOption, error) {
 	// In these cases, while we fallback to mTLS to istiod using the provisioned certs
 	// it would be ideal to keep using token plus k8s ca certs for control plane communication
 	// as the intention behind provisioned certs on k8s pods is only for data plane comm.
-	if sa.secOpts.ProvCert == "" {
-		// only if running in k8s pod
-		dialOptions = append(dialOptions, grpc.WithPerRPCCredentials(oauth.TokenSource{&fileTokenSource{
+	if !sa.secOpts.FileMountedCerts && sa.secOpts.ProvCert == "" {
+		dialOptions = append(dialOptions, grpc.WithPerRPCCredentials(oauth.TokenSource{TokenSource: &fileTokenSource{
 			sa.secOpts.JWTPath,
 			time.Second * 300,
 		}}))
 	}
 	return dialOptions, nil
 }
 
+// initCertificateWatches sets up  watches for the certs and resets upstream if they change.
+func (p *XdsProxy) initCertificateWatches(agent *Agent, stop <-chan struct{}) error {
```

Comment:
need checks to see if file certs are used, and if not return nil, without setting up go routine

Reply:
Not sure if this is done or not - but the check on L397 doesn't quite do this, we will always have a root cert, so it will always be used. Maybe that is a good idea though

Reply:
I did not know that root will always be there. I have corrected that check